telemetry/harvester_test.go

```diff
@@ -625,32 +625,67 @@ func TestRecordInvalidMetric(t *testing.T) {
 	}
 }
 
-func TestRequestRetryBody(t *testing.T) {
+func roundTripper(attempts int) roundTripperFunc {
```

Comment:
can we name it something more specific so it's more clear that this is to be used with only this test and benchmark?